src/Chapter2.hs

```diff
@@ -501,8 +518,11 @@ True
 >>> isThird42 [42, 42, 0, 42]
 False
 -}
-isThird42 = error "isThird42: Not implemented!"
-
+--isThird42 = error "isThird42: Not implemented!"
+isThird42 x
+    | length x<3=False
+    | (x!!2) == 42=True
```

Comment:
The `!!` function is not total, meaning it could fail on runtime. Even though you have a check on that, it is still better to avoid using that at all.
Also, the `length` function is costly here, and it won't work on infinite lists ⛓️ 
In this particular case, you could use the explicit pattern matching to solve this task 👌🏼